src/world/area_kkj/kkj_18/AF2FE0.c

```diff
@@ -1,17 +1,108 @@
 #include "kkj_18.h"
+#include "camera.h"
+
+extern s32 D_80241F10_AF4BB0;
+extern s32 D_80241F14_AF4BB4;
+extern s32 D_80244E78[];
+extern s32 tik_07_FlyingAI_JumpVels[];
 
 #include "world/common/GetNpcCollisionHeight.inc.c"
 
 #include "world/common/AddPlayerHandsOffset.inc.c"
 
+// Needs data migration, matching otherwise
+#ifdef NON_MATCHING
+ApiStatus func_8024056C_AF320C(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+
+    if (isInitialCall) {
+        D_80241F10_AF4BB0 = 0;
+    }
+
+    if (D_80241F10_AF4BB0 != 0) {
+        D_80241F10_AF4BB0 = 0;
+        evt_set_variable(script, *args++, D_80241F14_AF4BB4);
+        return ApiStatus_DONE2;
+    }
+
+    return ApiStatus_BLOCK;
+}
+#else
 INCLUDE_ASM(s32, "world/area_kkj/kkj_18/AF2FE0", func_8024056C_AF320C);
+#endif
+
+ApiStatus func_802405C0_AF3260(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+
+    D_80241F14_AF4BB4 = evt_get_variable(script, *args++);
+    D_80241F10_AF4BB0 = 1;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802405F8_AF3298(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32* ptr = (s32*)evt_get_variable(script, *args++);
+    s32 i;
+
+    if (ptr != NULL) {
+        for (i = 0; ptr[i] != 0; i++) {
+            tik_07_FlyingAI_JumpVels[i] = ptr[i];
+        }
+        tik_07_FlyingAI_JumpVels[i] = 0;
+    } else {
+        for (i = 0; i < 112; i++) {
+            tik_07_FlyingAI_JumpVels[i] = i + 16;
+            tik_07_FlyingAI_JumpVels[112] = 0;
+        }
```

Comment:
In cases like this, where we don't know the size of the array, would it be reasonable to assume `ARRAY_COUNT(tik_07_FlyingAI_JumpVels) = 112` and increase it later if needed?